clusterloader2/pkg/measurement/measurement_executor.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package measurement
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+)
+
+// ExecuteMeasurement executes a measurement, which can be a single measurement or a wrapper for multiple measurements.
+// It throws an error if both identifier and Instances has been supplied in the config.
+func ExecuteMeasurement(mm MeasurementManager, methodName string, identifier string, instances []api.MeasurementConfig, params map[string]interface{}) error {
+	if instances != nil && identifier != "" {
+		return fmt.Errorf("Only one of instances or identifier must be supplied")
+	}
+	if identifier != "" {
+		return executeSingleMeasurement(mm, methodName, identifier, params)
+	}
+	return executeWrapperMeasurement(mm, methodName, instances, params)
+}
+
+func executeSingleMeasurement(mm MeasurementManager, methodName string, identifier string, params map[string]interface{}) error {
+	err := mm.Execute(methodName, identifier, params)
+	if err != nil {
+		klog.Warning("Error while executing Measurement method - %s, identifier - %s, err - %v", methodName, identifier, err)
+	}
+	return err
+}
+
+func executeWrapperMeasurement(mm MeasurementManager, methodName string, instances []api.MeasurementConfig, params map[string]interface{}) error {
+
+	var wg wait.Group
+	errList := errors.NewErrorList()
+
+	for instance := range instances {
+
+		identifier := instances[instance].Identifier
+		measurementParams := instances[instance].Params
+
+		if measurementParams == nil {
+			measurementParams = params
+		} else {
+			for paramsKey := range params {
+				measurementParams[paramsKey] = params[paramsKey]
+			}
+		}
+
+		wg.Start(func() {
+			err := mm.Execute(methodName, identifier, measurementParams)
+			if err != nil {
+				errList.Append(fmt.Errorf("measurement call %s - %s error: %v", methodName, identifier, err))
+			}
+		})
+	}
+	wg.Wait()
+	if errList.IsEmpty() {
```

Comment:
do we need it?

Reply:
yes this is needed. Because errList is a list and if i don't send back a nil object, then because simple executor checks if err is nil, unwanted error lines get printed:

https://github.com/kubernetes/perf-tests/blob/b2afa608d8fd3134de6fad15350d469054c6f5f8/clusterloader2/pkg/test/simple_test_executor.go#L132

Reply:
OK - fine. Thanks for checking.